src/communication/network_interface_sr.c

```diff
@@ -7268,6 +7269,372 @@ slogtb_dump_trantable (THREAD_ENTRY * thread_p, unsigned int rid, char *request,
   db_private_free_and_init (thread_p, buffer);
 }
 
+/*
+ * smmon_get_server_info -
+ *
+ * return:
+ *
+ *   rid(in):
+ *   request(in):
+ *   reqlen(in):
+ */
+void
+smmon_get_server_info (THREAD_ENTRY * thread_p, unsigned int rid, char *request, int reqlen)
+{
+  char *buffer, *ptr;
+  int size = 0;
+  OR_ALIGNED_BUF (OR_INT_SIZE + OR_INT_SIZE) a_reply;
+  char *reply = OR_ALIGNED_BUF_START (a_reply);
+  int error = NO_ERROR;
+  MMON_SERVER_INFO server_info;
+
+  mmon_aggregate_server_info (server_info);
+
+  size += or_packed_string_length (server_info.name, NULL);
+  // Size of total_mem_usage
+  size += OR_INT64_SIZE;
+
+  buffer = (char *) db_private_alloc (thread_p, size);
+
+  if (buffer == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, size);
+      error = ER_OUT_OF_VIRTUAL_MEMORY;
+    }
+  else
+    {
+      ptr = or_pack_string (buffer, server_info.name);
+      ptr = or_pack_int64 (ptr, server_info.total_mem_usage);
+    }
+
+  if (error != NO_ERROR)
+    {
+      ptr = or_pack_int (reply, 0);
+      ptr = or_pack_int (ptr, error);
+      css_send_data_to_client (thread_p->conn_entry, rid, reply, OR_ALIGNED_BUF_SIZE (a_reply));
+    }
+  else
+    {
+      ptr = or_pack_int (reply, size);
+      ptr = or_pack_int (ptr, error);
+      css_send_reply_and_data_to_client (thread_p->conn_entry, rid, reply, OR_ALIGNED_BUF_SIZE (a_reply), buffer, size);
+    }
+  db_private_free_and_init (thread_p, buffer);
+}
+
+/*
+ * smmon_get_module_info -
+ *
+ * return:
+ *
+ *   rid(in):
+ *   request(in):
+ *   reqlen(in):
+ */
+void
+smmon_get_module_info (THREAD_ENTRY * thread_p, unsigned int rid, char *request, int reqlen)
+{
+  char *buffer, *ptr;
+  int size = 0;
+  OR_ALIGNED_BUF (OR_INT_SIZE + OR_INT_SIZE) a_reply;
+  char *reply = OR_ALIGNED_BUF_START (a_reply);
+  int error = NO_ERROR;
+  MMON_MODULE_INFO *module_info;
+  int module_index, module_count;
+
+  ptr = or_unpack_int (request, &module_index);
+
+  if (module_index == 0)
+    {
+      module_info = (MMON_MODULE_INFO *) malloc (sizeof (MMON_MODULE_INFO) * MMON_MODULE_LAST);
+      module_count = MMON_MODULE_LAST;
+    }
+  else
+    {
+      module_info = (MMON_MODULE_INFO *) malloc (sizeof (MMON_MODULE_INFO));
```

Comment:
Because `module_info` is used for a parameter of `mmon_aggregate_module_info()`, so I think that it is out-of-scope case. Is it okay to use `db_private_alloc()`?

Reply:
@Rudeus As I know, the scope of the db_private_alloc() is the duration of the thread job(request handle) execution. So, it is ok if the memory is freed before replying is done. Please let me know if there is any problem.

Reply:
I change it.